.github/workflows/release.yml

```
@@ -46,96 +88,117 @@ jobs:
+      - if: matrix.project == 'miner'
+        env:
+          GITHUB_TOKEN: ${{ github.token }}
+        run: make lotus-miner lotus-worker
+      - if: matrix.project == 'lotus' && runner.os == 'macOS'
         run: otool -hv lotus
-        working-directory: lotus
-      - env:
-          INPUTS_REF: ${{ inputs.ref }}
-        run: |
-          export GITHUB_REF=${INPUTS_REF:-$GITHUB_REF}
-          ../actions/scripts/version-check.sh ./lotus
-        working-directory: lotus
       - uses: actions/upload-artifact@v4
         with:
-          name: lotus-${{ matrix.os }}-${{ matrix.arch }}
+          name: lotus-${{ matrix.project }}-${{ matrix.os }}-${{ matrix.arch }}
           path: |
             lotus/lotus
             lotus/lotus-miner
             lotus/lotus-worker
   release:
-    name: Release [publish=${{ startsWith(inputs.ref || github.ref, 'refs/tags/') }}]
+    needs: [check, build]
+    if: needs.check.outputs.projects != '[]'
+    name: Release [publish=${{ github.event_name != 'pull_request' }}]
     permissions:
       # This enables the job to create and/or update GitHub releases
       contents: write
     runs-on: ubuntu-latest
-    needs: [build]
+    strategy:
+      fail-fast: false
+      matrix:
+        project: ${{ fromJSON(needs.check.outputs.projects) }}
     env:
-      PUBLISH: ${{ startsWith(inputs.ref || github.ref, 'refs/tags/') }}
+      PUBLISH: ${{ github.event_name != 'pull_request' }}
     steps:
-      - uses: actions/checkout@v4
-        with:
-          path: actions
       - uses: actions/checkout@v4
         with:
           submodules: 'recursive'
           fetch-depth: 0
-          path: lotus
-          ref: ${{ inputs.ref || github.ref }}
       - uses: actions/download-artifact@v4
         with:
-          name: lotus-Linux-X64
+          name: lotus-${{ matrix.project }}-Linux-X64
           path: linux_amd64_v1
       - uses: actions/download-artifact@v4
         with:
-          name: lotus-macOS-X64
+          name: lotus-${{ matrix.project }}-macOS-X64
           path: darwin_amd64_v1
       - uses: actions/download-artifact@v4
         with:
-          name: lotus-macOS-ARM64
+          name: lotus-${{ matrix.project }}-macOS-ARM64
           path: darwin_arm64
-      - uses: ./actions/.github/actions/install-go
-        with:
-          working-directory: lotus
+      - uses: ./.github/actions/install-go
       - uses: ipfs/download-ipfs-distribution-action@v1
         with:
           name: kubo
           version: v0.16.0
+      - id: meta
+        run: |
+          version="$(go run cmd/release/main.go --json ${{ matrix.project }} version | jq -r .msg)"
+          if [[ "${{ matrix.project}}" == "miner" ]]; then
+            tag="miner/$version"
+            latest="$(git tag -l 'miner/v*' | sort -V | tail -n1)"
+          elif [[ "${{ matrix.project}}" == "node" ]]; then
+            tag="$version"
+            latest="$(git tag -l 'v*' | sort -V | tail -n1)"
```

```suggestion
          if [[ "${{ matrix.project}}" == "node" ]]; then
            tag="$version"
            latest="$(git tag -l 'v*' | sort -V | tail -n1)"
          elif [[ "${{ matrix.project}}" == "miner" ]]; then
            tag="miner/$version"
            latest="$(git tag -l 'miner/v*' | sort -V | tail -n1)"
```

mind if we switch these to remain consistent in ordering?

---

I removed that part completely after testing.